test/decomposer/test_aba_decomposer.py

```diff
@@ -0,0 +1,44 @@
+from typing import Callable
+
+import numpy as np
+import pytest
+
+from opensquirrel.ir import BlochSphereRotation
+from opensquirrel.passes.decomposer import (
+    Decomposer,
+    XYXDecomposer,
+    XZXDecomposer,
+    YXYDecomposer,
+    YZYDecomposer,
+    ZXZDecomposer,
+    ZYZDecomposer,
+)
+from opensquirrel.passes.decomposer.general_decomposer import check_gate_replacement
+
+DECOMPOSER = [XYXDecomposer, XZXDecomposer, YXYDecomposer, YZYDecomposer, ZXZDecomposer, ZYZDecomposer]
+
+
+@pytest.mark.parametrize("decomposer_class", DECOMPOSER)
```

Comment:
Do we still need the `test_arbitrary` if we have the `test_full_sphere`?

Reply:
test_arbitrary is gone but I kept test_specific because it is very precise floats and also the test required for one of our stakeholders.